src/test/java/no/hvl/dat250/h2020/group5/controllers/PollControllerTest.java

```diff
@@ -0,0 +1,188 @@
+package no.hvl.dat250.h2020.group5.controllers;
+
+import no.hvl.dat250.h2020.group5.repositories.GuestRepository;
+import no.hvl.dat250.h2020.group5.repositories.PollRepository;
+import no.hvl.dat250.h2020.group5.repositories.UserRepository;
+import no.hvl.dat250.h2020.group5.repositories.VoteRepository;
+import no.hvl.dat250.h2020.group5.entities.*;
+import no.hvl.dat250.h2020.group5.enums.AnswerType;
+import no.hvl.dat250.h2020.group5.enums.PollVisibilityType;
+import no.hvl.dat250.h2020.group5.responses.PollResponse;
+import no.hvl.dat250.h2020.group5.responses.VotesResponse;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class PollControllerTest {
+
+    @LocalServerPort private int port;
+
+    @Autowired TestRestTemplate template;
+
+    @Autowired PollController PollController;
+
+    @Autowired PollRepository pollRepository;
+
+    @Autowired UserRepository userRepository;
+
+    @Autowired GuestRepository guestRepository;
+
+    @Autowired VoteRepository voteRepository;
+
+    private URL base;
+    private User user1;
+    private Poll poll1;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        pollRepository.deleteAll();
+        userRepository.deleteAll();
+        guestRepository.deleteAll();
+
+        user1 = new User();
+        user1.setUsername("Admin");
+        user1.setIsAdmin(true);
+
+        User user2 = new User();
+        user2.setUsername("Not admin");
+
+        Guest guest1 = new Guest();
+        guest1.setUsername("guest1");
+
+        poll1 = new Poll();
+        poll1.setQuestion("Question");
+        poll1.setPollOwner(user1);
+        poll1.setVisibilityType(PollVisibilityType.PUBLIC);
```

Comment:
Samme som i Guest. Hvis du gidder kan du lage en metode i Poll `public Poll question(...) {setQuestion(question); return this;}` og erstatte med 
```suggestion
        poll1 = new Poll().question("Question").pollOwner(user1).visibilityTyoe(PollVisibilityType.PUBLIC);
```